Relative imports broken in pylint 2.5.2

Pylint fails with relative imports within a package. Having a package b in directory a with three files:

`a/b/__init__.py`
<pre>from .one import ONE
from .two import TWO</pre>

`a/b/one.py`
<pre>ONE = 1</pre>

`a/b/two.py`
<pre>from .one import ONE
TWO = ONE + ONE</pre>

pylint 2.4.4 + astroid 2.3.3 do not generate any warning or error, which is as expected.

pylint 2.5.2 + astroid 2.4.1 generates an import-error:

<pre>$ export PYTHONPATH=/full/path/to/a
$ cd /full/path/to
$ pylint a
************* Module a.b
a/b/__init__.py:2:0: E0401: Unable to import 'a.b.two' (import-error)</pre>

Versions:

$ pylint --version
pylint 2.5.2
astroid 2.4.1
Python 3.8.3 (default, May 17 2020, 18:15:42) 
[GCC 10.1.0]

--- a/tests/lint/unittest_lint.py
+++ b/tests/lint/unittest_lint.py
@@ -39,6 +39,7 @@ from pylint.constants import (
 )
 from pylint.exceptions import InvalidMessageError
 from pylint.lint import PyLinter
+from pylint.lint.pylinter import MANAGER
 from pylint.lint.utils import fix_import_path
 from pylint.message import Message
 from pylint.reporters import text
@@ -912,6 +913,39 @@ def test_recursive_ignore(ignore_parameter, ignore_parameter_value) -> None:
     ):
         module = os.path.abspath(join(REGRTEST_DATA_DIR, *regrtest_data_module))
     assert module in linted_file_paths
+
+
+def test_relative_imports(initialized_linter: PyLinter) -> None:
+    """Regression test for https://github.com/PyCQA/pylint/issues/3651"""
+    linter = initialized_linter
+    MANAGER.clear_cache()  # Essential to reproduce the failure
+    with tempdir() as tmpdir:
+        create_files(["x/y/__init__.py", "x/y/one.py", "x/y/two.py"], tmpdir)
+        with open("x/y/__init__.py", "w", encoding="utf-8") as f:
+            f.write(
+                """
+\"\"\"Module x.y\"\"\"
+from .one import ONE
+from .two import TWO
+"""
+            )
+        with open("x/y/one.py", "w", encoding="utf-8") as f:
+            f.write(
+                """
+\"\"\"Module x.y.one\"\"\"
+ONE = 1
+"""
+            )
+        with open("x/y/two.py", "w", encoding="utf-8") as f:
+            f.write(
+                """
+\"\"\"Module x.y.two\"\"\"
+from .one import ONE
+TWO = ONE + ONE
+"""
+            )
+        linter.check(["x"])
+    assert not linter.stats.by_msg
 
 
 def test_import_sibling_module_from_namespace(initialized_linter: PyLinter) -> None:

--- a/tests/lint/unittest_lint.py
+++ b/tests/lint/unittest_lint.py
@@ -39,7 +39,6 @@ from pylint.constants import (
 )
 from pylint.exceptions import InvalidMessageError
 from pylint.lint import PyLinter
-from pylint.lint.pylinter import MANAGER
 from pylint.lint.utils import fix_import_path
 from pylint.message import Message
 from pylint.reporters import text
@@ -918,7 +917,6 @@ def test_recursive_ignore(ignore_parameter, ignore_parameter_value) -> None:
 def test_relative_imports(initialized_linter: PyLinter) -> None:
     """Regression test for https://github.com/PyCQA/pylint/issues/3651"""
     linter = initialized_linter
-    MANAGER.clear_cache()  # Essential to reproduce the failure
     with tempdir() as tmpdir:
         create_files(["x/y/__init__.py", "x/y/one.py", "x/y/two.py"], tmpdir)
         with open("x/y/__init__.py", "w", encoding="utf-8") as f:
@@ -944,7 +942,7 @@ from .one import ONE
 TWO = ONE + ONE
 """
             )
-        linter.check(["x"])
+        linter.check(["x/y"])
     assert not linter.stats.by_msg
 
 

